tests/test-pull-repeated.sh

```diff
@@ -67,3 +80,22 @@ ${CMD_PREFIX} ostree --repo=repo rev-parse main
 
 popd
 echo "ok repeated pull after 408s"
+
+# Test pulling a repo that gives 408s a lot of the time, with many network retries.
+rm ostree-srv httpd repo -rf
+setup_fake_remote_repo1 "archive" "${COMMIT_SIGN}" --random-408s=50
+
+pushd ${test_tmpdir}
+ostree_repo_init repo --mode=archive
+${CMD_PREFIX} ostree --repo=repo remote add --set=gpg-verify=false origin $(cat httpd-address)/ostree/gnomerepo
+
+# Using 8 network retries gives error rate of <0.5%, when --random-408s=50
+if ${CMD_PREFIX} ostree --repo=repo pull --mirror origin --network-retries=8 main 2>err.txt; then
+   echo "Success with big number of network retries"
+fi
```

Comment:
Minor: usually for tests we don't expect to fail, we just run it directly instead of wrapping it around an if statement. Because we're running with `set -e`, any command that fails will abort the test. See: http://redsymbol.net/articles/unofficial-bash-strict-mode/.

Not worth a respin, just FYI :)

Reply:
Actually, thinking on this again, wrapping it around an if statement means that we'll silently continue even if it fails. So we do need to change this.

Want to post a follow-up PR once this one is merged? :)

Reply:
Ah I see - previously I tried giving `--network-retries=0` to make it fail, and it did stop and fail, but for the wrong reason: it stopped at `${CMD_PREFIX} ostree --repo=repo rev-parse main` because nothing had been pulled in. Just tried running it directly now, it stopped at the pull command as it should. Will post a follow-up PR.

Thanks for pointing that out! Will read over the link.